source-code/app/src/main/java/org/buildmlearn/toolkit/activity/TemplateEditor.java

```diff
@@ -163,6 +163,9 @@ private void save_apk() {
         String aliasName = getString(R.string.alias_name);
         String aliaspassword = getString(R.string.alias_password);
         KeyStoreDetails keyStoreDetails = new KeyStoreDetails(keyPassword, aliasName, aliaspassword);
+        if(saveProject().equals("File already exists")){
```

Comment:
You can add all the exceptions from saveProject here. In all those cases, apk should not be created